### ISSUE TITLE
Load Agent Skills at Boot

### ISSUE THREAD

**Issue by geowa4** [585-software/medera#10]
Each Agent has many skills. When the Agent starts. it must load these skills definitions from a YAML configuration file. Each skill is a verb, noun, description, and a command. The verb-noun pair is what is entered in Slack (e.g. `list-homedirectory`), and the command is the executable with its arguments. See below for an example configuration file.

```
---
skills:
  - description: Lists files in a directory
    verb: list
    noun: homedirectory
    command: /bin/ls ~
```

**Comment by dantswain** [comment]
Are you set on YAML? Elixir sort of eschews YAML.

**Comment by geowa4** [comment]
No, not really. As long as it's easy to read and write for your random administrator, it doesn't really matter to me.

**Comment by dantswain** [comment]
Cool.  XML?  :trollface: 

I say JSON unless you have a better idea.

**Comment by geowa4** [comment]
Yeah, as a POC, I can be fine with that. In the future, I might wish to support a language that is a superset of JSON.

**Comment by dantswain** [comment]
Loading skills at boot is trivial, but I'm taking "those skills actually work and the code isn't horrible" as an exit condition.

**Comment by dantswain** [comment]
@geowa4 From a PM standpoint are you comfortable with closing this (and opening some new tasks to do refinement) as of #21 or do you want it to remain open until those are refined a bit?

**Comment by dantswain** [comment]
Closed by #21
